lib/Sema/CSDiag.cpp

```diff
@@ -1426,10 +1413,6 @@ namespace {
     // we go digging through failed constraints, and expect their locators to
     // still be meaningful.
     ~ExprTypeSaverAndEraser() {
-      for (auto CSE : CollectionSemanticExprs)
```

Comment:
I love that ExprTypeSaverAndEraser becomes simpler as well (ideally it wouldn't exist; there's a lot of refactoring before we can do that though!)